src/genie/libs/parser/nxos/show_interface.py

```diff
@@ -3679,15 +3679,15 @@ def cli(self, interface="", output=None):
         # --------------------------------------------------------------------------------
         # Eth1/1        KeepAlive          connected routed    full    10G     10g
         # Eth1/2        AOTLXPRISS10004    connected trunk     full    10G     10g
-        # mgmt0         --                 connected routed    full    1000    --
+        # mgmt0         --                 connected routed    full    1000q
         # Po1           VPC_PeerLink       connected trunk     full    40G     --
-        # Vlan366       BigData            connected routed    auto    auto    --
+        # Vlan366       BigData            connected routed    auto    auto
         # Eth101/1/10   DO-HYPER-03        connected 101       full    a-1000
         # Lo0            --                  connected  routed     auto     --       --
         p1 = re.compile(r'(?P<interface>(\S+))\s+(?P<name>(\S+))?\s'
                         r'+(?P<status>(\S+))?\s+(?P<vlan>(\S+))'
                         r' +(?P<duplex_code>(\S+))\s'
-                        r'+(?P<port_speed>(\S+))(\s+(?P<type>(\S+)))?$')
+                        r'+(?P<port_speed>(\S+))(\s*(?P<type>(\S*)))?$')
```

Comment:
This should be covered by the `?` at the end I would think. I suppose if it works though and no tests are breaking then this is fine

Reply:
The `?` does not work properly for empty `type`. 
Columns are wrongly parsed in json output.
Example (line `3690`) `Eth102/1/3    zzz (Eth1, test_st connected 205       full    a-1000`
```
'Ethernet102/1/3': {'duplex_code': 'connected',
      'name': 'zzz',
      'port_speed': '205',
      'status': '(Eth1,',
      'type': 'full    a-1000',
      'vlan': 'test_st'},
```

Reply:
Makes sense, thank you